src/HieDb/Utils.hs

```diff
@@ -83,22 +83,20 @@ makeNc = do
   uniq_supply <- mkSplitUniqSupply 'z'
   return $ initNameCache uniq_supply []
 
--- | Recursively search for .hie files in given directory
+-- | Recursively search for .hie and .hie-boot files in given directory
 getHieFilesIn :: FilePath -> IO [FilePath]
 getHieFilesIn path = do
-  exists <- doesPathExist path
```

Comment:
This `doesPathExist` check seems to be redundant, because we don't do anything differently that we're already doing based on `doesFileExist` and `doesDirectoryExist` checks within this function.